contracts/RemovalUtils.sol

```diff
@@ -0,0 +1,214 @@
+uint256 constant _COUNTRY_CODE_OFFSET = 26;
+uint256 constant _ADMIN1_CODE_OFFSET = 24;
+uint256 constant _ADDRESS_OFFSET = 4;
+uint256 constant _SUBID_OFFSET = 0;
+
+/**
+ * @dev Library encapsulating the logic around encoding and decoding removal token ids.
+ *
+ * The token ids used for a given ERC1155 token in Removal.sol encode information about the carbon removal
+ * in the following format(s), where the first byte encodes the format version:
+ *
+ * Version 0:
+ * [1byte][1byte][--2 bytes--][--2 bytes--][--2 bytes--][----------- 20 bytes------------- ][------4 bytes------]
+ * tokIdV--meth&v---vintage------country------admin1------------ supplier address --------------subidentifier--
+ *
+ *
+ */
+library RemovalUtils {
+  function createRemovalId(bytes calldata removalData)
+    internal
+    pure
+    returns (uint256)
+  {
+    uint256 idVersion = abi.decode(removalData, (uint256));
+    require(idVersion == 0, "Unsupported removal token id version");
+
+    UnpackedRemovalIdV0 memory params = abi.decode(
+      removalData,
+      (UnpackedRemovalIdV0)
+    );
+    bytes memory countryBytes = bytes(params.country);
+    bytes memory admin1Bytes = bytes(params.admin1);
+    require(countryBytes.length == 2, "Country code must be two characters");
+    require(admin1Bytes.length == 2, "Admin1 code must be two characters");
+    // TODO: additional field validation & tests for validation logic
+    uint256 methodologyData = (params.methodology << 4) |
+      params.methodologyVersion;
+
+    return
+      (params.idVersion << (_ID_VERSION_OFFSET * _BITS_PER_BYTE)) |
+      (methodologyData << (_METHODOLOGY_DATA_OFFSET * _BITS_PER_BYTE)) |
+      (params.vintage << (_VINTAGE_OFFSET * _BITS_PER_BYTE)) |
+      ((uint256(uint8(countryBytes[0])) <<
+        (_COUNTRY_CODE_OFFSET * _BITS_PER_BYTE + _BITS_PER_BYTE)) |
+        (uint256(uint8(countryBytes[1])) <<
+          (_COUNTRY_CODE_OFFSET * _BITS_PER_BYTE))) |
+      ((uint256(uint8(admin1Bytes[0])) <<
+        (_ADMIN1_CODE_OFFSET * _BITS_PER_BYTE + _BITS_PER_BYTE)) |
+        (uint256(uint8(admin1Bytes[1])) <<
+          (_ADMIN1_CODE_OFFSET * _BITS_PER_BYTE))) |
+      (uint256(uint160(params.supplierAddress)) <<
+        (_ADDRESS_OFFSET * _BITS_PER_BYTE)) |
+      (params.subIdentifier << (_SUBID_OFFSET * _BITS_PER_BYTE));
+  }
+
+  function unpackRemovalId(uint256 removalId)
+    internal
+    pure
+    returns (UnpackedRemovalIdV0 memory)
+  {
+    uint256 version = versionFromRemovalId(removalId);
+    uint256 methodology = methodologyFromRemovalId(removalId);
+    uint256 methodologyVersion = methodologyVersionFromRemovalId(removalId);
+    uint256 vintage = vintageFromRemovalId(removalId);
+    string memory country = countryCodeFromRemovalId(removalId);
+    string memory admin1 = admin1CodeFromRemovalId(removalId);
+    address supplierAddress = supplierAddressFromRemovalId(removalId);
+    uint256 subIdentifier = subIdentifierFromRemovalId(removalId);
```

Comment:
:+1: I think that's a more future proof name.  I would splatter lots of comments around that link it back to profile.